brave/brave-core
# [ads] Implement brave://ads-internals

Security/privacy review issue: https://github.com/brave/reviews/issues/1794

Screenshots of the brave://ads-internals page are available here:
https://github.com/brave/brave-browser/issues/40952#issuecomment-2477221118


Resolves https://github.com/brave/brave-browser/issues/40952



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

### The functionality should be tested for iOS, Android and Desktop

### Test case 1
- Fresh install
- Open brave://ads-internals
- Confirm that there are no conversion-url-pattern records on the page
- Open staging search `search.brave.software` in a new tab
- Trigger search result ad for a staging server with the giraffe request
- Click search result ad
- Switch back to the tab with brave://ads-internals
- Click `Refresh` button
- Confirm that there is a conversion-url-pattern record on the page
- Click `Clear Ads Data` button
- Confirm that there are no conversion-url-pattern records on the page
- Join Brave Rewards
- Confirm that `Clear Ads Data` button becomes hidden
- Open staging search `search.brave.software` in a new tab
- Trigger search result ad for a staging server with the giraffe request
- Switch back to the tab with brave://ads-internals
- Click `Refresh` button
- Confirm that there is a conversion-url-pattern record on the page

### Test case 2
- Fresh install
- Open new Private Window (or Tab)
- Open `brave://ads-internals`
- Click `Refresh` and `Clear Ads Data` buttons
- EXPECTATION: Browser didn't crash and `brave://ads-internals` is shown correctly.
